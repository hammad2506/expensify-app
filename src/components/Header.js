import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogoutUser } from '../actions/auth';


export const Header = ( {logoutUser} ) => {
    
    const onLogoutClick = () => { 
        logoutUser()
     };
    
    return (
        <header className='header'>
            <div className='content-container'>
                    <div className='header__content'>
                        <Link className='header__title' to='/dashboard'>
                            <h1>Expensify</h1>
                        </Link>
                        <button className='button button--link' onClick={onLogoutClick}>Log out</button>
                    </div>
            </div>
        </header>
        
     );
};

const mapDispatchToProps = (dispatch) => ({
    logoutUser : () => dispatch(startLogoutUser())
});

export default connect(undefined, mapDispatchToProps)(Header);