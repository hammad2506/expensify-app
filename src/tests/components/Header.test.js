import React from 'react';
import Header from '../../components/Header';
import { shallow } from 'enzyme';

test("Testing Header Component", () => {
    const wrapper =  shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
});