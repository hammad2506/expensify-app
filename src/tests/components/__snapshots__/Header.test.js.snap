// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header Component 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Log out 
  </button>
  <NavLink
    activeClassName="activeLink"
    aria-current="page"
    exact={true}
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="activeLink"
    aria-current="page"
    to="/add"
  >
    Add
  </NavLink>
  <NavLink
    activeClassName="activeLink"
    aria-current="page"
    to="/help"
  >
    Help
  </NavLink>
</div>
`;
